snap/snapcraft.yaml

```diff
@@ -1,56 +1,83 @@
 name: matter-pi-gpio-commander
-version: "0.1"
-summary: Matter Raspberry Pi GPIO output commander
-description: |
-  TBA
-license: Apache-2.0
-confinement: strict
-base: core22
+version: "0.2"
+summary: Raspberry Pi GPIO as a Matter lighting app
+description: Refer to https://snapcraft.io/matter-pi-gpio-commander
+
 grade: devel
+confinement: strict
 
+base: core22
 architectures:
   - build-on: arm64
   - build-on: armhf
+  # - build-on: amd64
 
 layout:
   /mnt:
-    bind: $SNAP_DATA/mnt
+    bind: $SNAP_COMMON/mnt
 
 parts:
-  build-res:
-    plugin: dump
-    source: build
+  wiring-pi:
+    plugin: nil
+    source: https://github.com/WiringPi/WiringPi.git
+    source-depth: 1
+    source-tag: 2.61-1
     override-build: |
-      cp requirements.txt $CRAFT_STAGE/
+      WIRINGPI_SUDO="" ./build
+
+      mkdir -p $CRAFT_PART_INSTALL/lib
+      cp -v /usr/local/lib/libwiringPi.so* $CRAFT_PART_INSTALL/lib/
+    build-packages: [git, gcc, g++]
+
+  test-blink:
+    after: [wiring-pi]
+    plugin: nil
+    source: .
+    override-build: |
+      g++ -Wall -o test-blink test-blink.cpp -lwiringPi
+      mkdir -p $CRAFT_PART_INSTALL/bin
+      cp test-blink $CRAFT_PART_INSTALL/bin/
+  
   connectedhomeip:
-    after: [build-res]
     plugin: nil
-    override-build: |   
-      git clone https://github.com/project-chip/connectedhomeip.git --depth=1 .
+    source: https://github.com/project-chip/connectedhomeip.git
+    source-depth: 1
+    source-tag: v1.0.0.2
+    source-submodules: []
```

Comment:
Is this obsolete? As it is empty, no submodules are fetched here.

Reply:
That's done by purpose so that we shallow clone the submodules in override-pull.